client/state/analytics/actions.js

```diff
@@ -147,3 +147,57 @@ const withClientId = actionCreator => ( ...args ) => ( dispatch, getState ) => {
 
 export const recordTracksEventWithClientId = withClientId( recordTracksEvent );
 export const recordPageViewWithClientId = withClientId( recordPageView );
+
+/**
+ * Dispatches the specified Redux action creator once enhancers have been applied to the result of its call. This
+ * function can be seen as an alternative to Redux middlewares with a more focused/local scope.
+ */
+export const withEnhancers = ( actionCreator, enhancers ) => ( ...args ) => ( dispatch, getState ) => {
+	const action = actionCreator( ...args );
+
+	if ( typeof action !== 'object' ) {
+		throw new Error( 'withEnhancers only works with an action creator that returns a plain action object' );
+	}
+
+	if ( ! Array.isArray( enhancers ) ) {
+		enhancers = [ enhancers ];
+	}
+
+	return dispatch( enhancers.reduce(
+		( result, enhancer ) => {
+			return merge( result, enhancer( result, getState ) );
+		},
+		action
+	) );
+};
```

Comment:
This is heavily inspired from @yurynix's work on [`withClientId()`](https://github.com/Automattic/wp-calypso/blob/abcdf173859e3c07ade099cedb7ab1521a08339b/client/state/analytics/actions.js#L124-L146).

Reply:
This is a general purpose function. I suspect we could move it outside from the analytics actions.